File: apps/web/types/AuthResponseDto.ts
/**
 * Shape of the JSON payload returned by every successful auth endpoint.
 *
 * Mirrors the backend's `AuthResponseDto`: the short-lived access token plus
 * its remaining lifetime in seconds. The refresh token travels separately as
 * an HTTP-only cookie and is therefore not part of this interface.
 */
export interface AuthResponseDto {
  accessTk: string;
  expiresIn: number;
}

File: apps/backend/src/main/java/com/authkit/backend/auth/dto/responses/AuthResponseDto.java
package com.authkit.backend.auth.dto.responses;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Response payload returned by every successful auth endpoint.
 *
 * <p>Exposes only the short-lived access token (the refresh token travels in a
 * separate HTTP-only cookie) and its remaining lifetime in seconds so the
 * client can schedule a silent refresh before expiry.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponseDto {

	/** The signed, short-lived JWT the client sends as {@code Authorization: Bearer ...}. */
	private String accessTk;

	/** Remaining access-token lifetime in seconds (900 = 15 minutes). */
	private Long expiresIn;
}
